crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -457,19 +457,54 @@ impl GenericQueryBuilder<Pg> for PgQueryBuilder {
         }
 
         // Filters on the package and/ or module that emitted some event
-        if let Some(p) = filter.emitting_package {
-            query = query.filter(events::dsl::package.eq(p.into_vec()));
+        if let Some(pm) = filter.emitting_module {
+            let parts: Vec<_> = pm.splitn(2, "::").collect();
 
-            if let Some(m) = filter.emitting_module {
-                query = query.filter(events::dsl::module.eq(m));
+            if parts.iter().any(|&part| part.is_empty()) {
+                return Err(DbValidationError::InvalidType(
+                    "Empty strings are not allowed".to_string(),
```

Comment:
This error message could be clearer -- e.g. 

- Where is the empty string? (in a component of the module filter)
- What are the correct formats that this filter accepts?
- Share the string that caused the error in the first place.

Reply:
Settled on the below to answer your questions.
```
#[derive(thiserror::Error, Debug)]
pub enum TypeFilterError {
    #[error("Invalid format in '{0}' - if '::' is present, there must be a non-empty string on both sides. Expected format like '{1}'")]
    MissingComponents(String, &'static str),
    #[error("Invalid format in '{0}' - value must have {1} or fewer components. Expected format like '{2}'")]
    TooManyComponents(String, u64, &'static str),
}
```
Where the format is something like `package[::module]` or `package[::module[::type[<type_params>]]]`

1. since we return an error on any portion of the segmented string that is "empty", I think the MissingComponents error variant is still clear on what the user's error was without them having to whack-a-mole - it makes it clear that any usage of "::" requires a preceding and succeeding non-empty value
2. Provides the expected format, using square brackets to indicate optionality
3. Also shares the string that caused the error in the first place